### ISSUE TITLE
Home Assistant integration installation failed

### ISSUE THREAD

**Issue by robertwebbe** [mikewoudenberg/homeassistant-loqed#10]
Hi @mikewoudenberg,

I've tried to install it in Home Assistant and after entering the personal access token, it stated that the installion was succesfull.
But when I'm looking at the integratyions, I see "Failed to set up":

![Picture 1](https://tweakers.net/i/POz6vZj6dBFTUP5PElBtMjNF5XU=/full-fit-in/4000x4000/filters:no_upscale():fill(white):strip_exif()/f/image/xyy9JYVHu3kozv7l5i1SMA70.png?f=user_large)

![Picture 2](https://tweakers.net/i/48onMD0rh44zWcFvNm2LZ2TyL4s=/full-fit-in/4000x4000/filters:no_upscale():fill(white):strip_exif()/f/image/IIEeKSrFAyBKFHkmXNQHn1Wr.png?f=user_large)

I've found this data in the log files:
```
Logger: homeassistant.config_entries
Source: helpers/network.py:200
First occurred: 23:11:28 (1 occurrences)
Last logged: 23:11:28

Error setting up entry LOQED Touch Smart Lock for loqed
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 387, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/loqed/__init__.py", line 37, in async_setup_entry
    await coordinator.ensure_webhooks()
  File "/config/custom_components/loqed/coordinator.py", line 118, in ensure_webhooks
    webhook_url = webhook.async_generate_url(self.hass, webhook_id)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/webhook/__init__.py", line 92, in async_generate_url
    get_url(hass, prefer_external=True, allow_cloud=False),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/network.py", line 200, in get_url
    raise NoURLAvailableError
homeassistant.helpers.network.NoURLAvailableError
```
```
Logger: homeassistant.config_entries
Source: components/webhook/__init__.py:53
First occurred: 23:11:47 (1 occurrences)
Last logged: 23:11:47

Error setting up entry LOQED Touch Smart Lock for loqed
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 387, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/loqed/__init__.py", line 37, in async_setup_entry
    await coordinator.ensure_webhooks()
  File "/config/custom_components/loqed/coordinator.py", line 115, in ensure_webhooks
    webhook.async_register(
  File "/usr/src/homeassistant/homeassistant/components/webhook/__init__.py", line 53, in async_register
    raise ValueError("Handler is already defined!")
ValueError: Handler is already defined!
```

I guess it might be something with Home Assistant Cloud.
Can you please take a look into it?

**Comment by mikewoudenberg** [comment]
Hey thanks for reporting. Looks like the integration cannot determine a reachable URL for the webhook that gets registered on the lock's bridge for communicating lock state updates. I'll have to investigate this a bit as I leverage the `webhook.async_generate_url` call here provided by the Home Assistant platform. I'll let you know if I find something.

**Comment by mikewoudenberg** [comment]
I've pushed a new version of the component on the main branch which uses the most flexible way possible to determine the home assistant instance URL favouring local connections. Hope this resolves your issue. Note that I also created a tagged 0.9.0 version which doesn't contain this fix but will still work for users that have a different setup. I;ve done this because I don't have Nabu Casa to test this.
